x-pack/plugins/alerting/server/routes/health.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  IRouter,
+  RequestHandlerContext,
+  KibanaRequest,
+  IKibanaResponse,
+  KibanaResponseFactory,
+  ElasticsearchServiceSetup,
+} from 'kibana/server';
+import { isApiKeyDisabledError, isSecurityPluginDisabledError } from './lib/error_handler';
+import { AlertingFrameworkHealth } from '../types';
+
+export function healthRoute(router: IRouter, elasticsearch: ElasticsearchServiceSetup) {
+  const clusterClient = elasticsearch.createClient('alertingSecurity', {
+    plugins: [elasticsearchClientPlugin],
+  });
+
+  router.get(
+    {
+      path: '/api/alert/_health',
+      validate: false,
+    },
+    router.handleLegacyErrors(async function(
+      context: RequestHandlerContext,
+      req: KibanaRequest<any, any, any, any>,
+      res: KibanaResponseFactory
+    ): Promise<IKibanaResponse<any>> {
+      try {
+        const scopedClusterClient = clusterClient.asScoped(req);
+
+        const {
+          canGenerateApiKeys,
+        }: AlertingFrameworkHealth = await scopedClusterClient
+          .callAsCurrentUser('alertingSecurity.canGenerateApiKeys', { owner: true })
```

Comment:
As I'm testing this, I just realized this doesn't work because not every user has access to the API keys API (requires `manage_api_key`). `callAsInternalUser` won't work either because the internal user doesn't have the special privilege. There are some changes coming that will allow kibana to create and invalidate API keys, maybe there's something we can do with that 🤔

Reply:
We've chatted over zoom and seems there was a major issue with how we (and Stack Monitoring, for that matter) check that ApiKeys work.

I'm looking into whether there's another way to check this as we speak.

Reply:
To avoid requiring changes from other teams and with the discovery of `/_xpack/usage` API. I think we're better off just pinging that API in 7.7 to see if SSL is enabled/disabled for HTTP and if security is enabled/disabled. Telemetry uses this endpoint so must be good enough even though it has its downsides (runs only on master ES node, etc).

https://www.elastic.co/guide/en/elasticsearch/reference/current/usage-api.html